datumaro/plugins/mnist_csv_format.py

```diff
@@ -0,0 +1,169 @@
+        meta = []
+        if osp.isfile(metafile):
+            with open(metafile, 'r', encoding='utf-8') as f:
+                meta = f.readlines()
+
+        for i, data in enumerate(annotation_table):
+            data = data.split(',')
+            item_anno = []
+            try:
+                label = int(data[0])
+            except ValueError:
+                label = None
+            if label != None:
+                item_anno.append(Label(label))
+
+            if 0 < len(meta):
+                meta[i] = meta[i].strip().split(',')
+
+            # support for single-channel image only
+            image = None
+            if 1 < len(data):
+                if 0 < len(meta) and 1 < len(meta[i]):
+                    image = np.array([int(pix) for pix in data[1:]],
+                        dtype='uint8').reshape(int(meta[i][-2]), int(meta[i][-1]))
+                else:
+                    image = np.array([int(pix) for pix in data[1:]],
+                        dtype='uint8').reshape(28, 28)
+
+            if 0 < len(meta) and len(meta[i]) in [1, 3]:
+                i = meta[i][0]
+
+            items[i] = DatasetItem(id=i, subset=self._subset,
+                image=image, annotations=item_anno)
+        return items
+
+class MnistCsvImporter(Importer):
+    @classmethod
+    def find_sources(cls, path):
+        return cls._find_sources_recursive(path, '.csv', 'mnist_csv',
+            file_filter=lambda p: not osp.basename(p).startswith('meta'))
+
+class MnistCsvConverter(Converter):
+    DEFAULT_IMAGE_EXT = '.png'
+
+    def apply(self):
+        for subset_name, subset in self._extractor.subsets().items():
+            data = []
+            item_ids = {}
+            image_sizes = {}
+            for item in subset:
+                anns = [a.label for a in item.annotations
+                    if a.type == AnnotationType.label]
+                label = None
+                if anns:
+                    label = anns[0]
+
+                if item.has_image and self._save_images:
+                    image = item.image
+                    if not image.has_data:
+                        data.append([label, None])
+                    else:
+                        if image.data.shape[0] != MnistCsvPath.IMAGE_SIZE or \
+                                image.data.shape[1] != MnistCsvPath.IMAGE_SIZE:
+                            image_sizes[len(data)] = [image.data.shape[0],
+                                image.data.shape[1]]
+                        image = image.data.reshape(-1).astype(np.uint8).tolist()
+                        image.insert(0, label)
+                        data.append(image)
+                else:
+                    data.append([label])
+
+                if item.id != str(len(data) - 1):
+                    item_ids[len(data) - 1] = item.id
+
+            anno_file = osp.join(self._save_dir, 'mnist_%s.csv' % subset_name)
```

Comment:
I think we need to create an save directory if it doesn't exist. What do you think?

Reply:
It is guaranteed to exist at this point in all practical use cases, but if this class is used directly, this can be a problem.

Reply:
But when I try to export mnist dataset to unexisting directory with command  `datum export -f mnist_csv -o ./newdir`, I get an error `No such file or directory: '<project_path>/newdir/mnist_train.csv'`. And for other formats, this command works. Or am I not exporting correctly?

Reply:
Yes, you're right, this error appears. It will appear in all formats that have no subfolders. This needs to be fixed, but I'm not sure what needs to be created in each format separately. Probably this can be done in [converter](https://github.com/openvinotoolkit/datumaro/blob/962ade03232aefba40c7adef92635cb6557adc59/datumaro/components/converter.py#L41) for all formats.